addons/position_manager/position_manager.lua

```diff
@@ -0,0 +1,83 @@
+--Copyright (c) 2020, Lili
+--All rights reserved.
+
+--Redistribution and use in source and binary forms, with or without
+--modification, are permitted provided that the following conditions are met:
+
+--    * Redistributions of source code must retain the above copyright
+--      notice, this list of conditions and the following disclaimer.
+--    * Redistributions in binary form must reproduce the above copyright
+--      notice, this list of conditions and the following disclaimer in the
+--      documentation and/or other materials provided with the distribution.
+--    * Neither the name of position_manager nor the
+--      names of its contributors may be used to endorse or promote products
+--      derived from this software without specific prior written permission.
+
+--THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+--ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+--WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+--DISCLAIMED. IN NO EVENT SHALL Lili BE LIABLE FOR ANY
+--DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+--(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+--LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+--ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+--(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+--SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+_addon.name = 'position_manager'
+_addon.author = 'Lili'
+_addon.version = '1.0.0'
+_addon.command = 'pm'
+
+if not windower.file_exists(windower.windower_path .. '\\plugins\\WinControl.dll') then
+    print('position_manager: error - Please install the WinControl plugin in the launcher.')
+    windower.send_command('lua u position_manager')
+    return
+end
+
+config = require 'config'
+
+default = {
+    x = 0,
+    y = 0,
+}
+
+settings = config.load(default)
+
+function move(settings)
+    windower.send_command('load wincontrol')
+    coroutine.sleep(0.5)
+    windower.send_command('wincontrol move %s %s':format(settings.x,settings.y))
+    coroutine.sleep(0.5)
+    windower.send_command('unload wincontrol')
+end
```

Comment:
I am *certain* this will make users complain. If someone wants to have *WinControl* loaded all the time this will unload it again, because it performs no checks (which is not your fault, it is only really possible with an ugly hack).

I think expecting the plugin to be loaded is reasonable in this case. Would be faster, and would not hurt anyone's current workflow.

Reply:
That's a good point, albeit once position_manager is loaded there's no longer any reason to keep wincontrol loaded at all times, and if one wants to do some testing they can always reload the plugin manually. In any case, I'll remove the unload command.

I still would like to avoid situations where the command is sent but nothing happens, so what do you think about sending a `load wincontrol` command once on addon load, rather than each time a window is moved?

Reply:
I would much prefer it to loading it for every move, though I still feel it is not *necessary*, but in this case I feel my personal preference for clean design is getting in the way of practicality, so I will completely defer to you :D

Reply:
Oh and I forgot to say, WinControl also has some other features, resizing and setting it to the foreground. I did not really think that anyone used those, but I also did not think anyone used WinControl at all, so shows how much I know :X

Reply:
I had no idea those functions of WinControl existed myself! The setting to foreground might be useful to reimplement what FFXISwitchMon used to do, but I digress.

I think I prefer having at least one load onload: if the addon has been enabled in the launcher, there's a reasonable expectation for it to _do something_. With the plugin not loaded, it would fail entirely silently, which would lead users to come asking "hey why doesn't it work", and I'd like to avoid that. At worst it'll blurt out one "plugin already loaded" error, but it's only once and pretty early in the login process, so I feel it wouldn't be too bothersome, and it's preferable imo to failing entirely silently.